builtin/game/falling.lua

```diff
@@ -146,13 +146,29 @@ end
 
 local function drop_attached_node(p)
 	local n = core.get_node(p)
+	local node_meta = core.get_meta(p)
+	local meta_fields = node_meta and node_meta:to_table().fields
 	core.remove_node(p)
 	for _, item in pairs(core.get_node_drops(n, "")) do
 		local pos = {
 			x = p.x + math.random()/2 - 0.25,
 			y = p.y + math.random()/2 - 0.25,
 			z = p.z + math.random()/2 - 0.25,
 		}
+		-- If it drops itself, preserve its metadata.
+		if meta_fields then
+			-- item is a table in 0.4.16, but a string in 0.5 - handle both.
```

Comment:
that's not the reason why this is here, this code will never once be used in 0.4.16 (it will be merged into 0.5)
the actual reason is that you can't rely on `get_node_drops` returning a string / table / whatever
IMO you don't need a comment describing this